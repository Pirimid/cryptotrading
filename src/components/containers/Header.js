import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as actions from '../../actions/headerActions';

export class Header extends React.Component {

 changePair = (pair) => {
  return (() => {
    this.props.actions.updateCurrencyPair( pair );
  }).bind(this);
};

  render() {
	return (
	  <div className="nav" >
		  <div className="navItems navlogo-container"><span className="navlogo" >Crypto Trade</span></div>
		  <div className="navItems pair-selection-container p-r has-dropdown">
			  <div className="pair-selection">
				  <div className="pair nav-title">{this.props.currentPair.unit1}/{this.props.currentPair.unit2}</div>
				  <div className="selection nav-sm-title">Select product</div>
			  </div>
			  <i className="fa arrow fa-chevron-down" aria-hidden="true"></i>
			  <div className="nav-dropdown">
					<h5 className="nav-title">BITCOIN(BTC)</h5>
					<ul className="nav-dropdown-list">
                        {this.props.appData.availablePairs.pairCategory1.map((object, i) => (
    						<li key={i} onClick={this.changePair(object)} >
    						    <a >
    						        <span>{object.unit1} / {object.unit2}</span>
    						        <span>{object.symbolUnit2} {object.lastTradePrice}</span>
    						        <span className={ object.last24HourPrice > 0 ? "up" : "down"}>{object.last24HourPrice}%</span>
                                </a>
                            </li>
                        ))}
					</ul>
			  </div>
		  </div>
		  <div className="navItems">
			  <div className="pair-selection">
				  <div className="pair nav-title">{this.props.currentPair.lastTradePrice} {this.props.currentPair.unit1}</div>
				  <div className="selection nav-sm-title">Last trade price</div>
			  </div>
		  </div>
		  <div className="navItems">
			  <div className="pair-selection">
				  <div className="pair nav-title green">{this.props.currentPair.last24HourPrice} %</div>
				  <div className="selection nav-sm-title">24 hour price</div>
			  </div>
		  </div>
		  <div className="navItems">
			  <div className="pair-selection">
				  <div className="pair nav-title">{this.props.currentPair.last24HourVolume} {this.props.currentPair.unit2}</div>
				  <div className="selection nav-sm-title">24 hour volume</div>
			  </div>
		  </div>
	  </div>
	);
  }
}

Header.propTypes = {
  actions: PropTypes.object.isRequired,
  currentPair: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
	currentPair: state.currentPair,
	appData: state.appData
  };
}

function mapDispatchToProps(dispatch) {
  return {
	actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
