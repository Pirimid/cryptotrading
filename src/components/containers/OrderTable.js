import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {Grid, Row, Col } from 'react-bootstrap';
import * as actions from '../../actions/fuelSavingsActions';

export class OrderTable extends React.Component {

  render() {
		return (
			<div>
				<h2 className="card-title dark-bg">Order Book</h2>
				<div className="clearfix"></div>
				<table className="table l-g-table">
					<thead>
						<tr>
							<th>Size</th>
							<th>Filled(BTC)</th>
							<th>Price</th>
							<th>Time</th>
							<th>Status</th>
						</tr>
					</thead>
					<tbody>
						<tr>
							<td>1</td>
							<td>Mark</td>
							<td>Otto</td>
							<td>sd</td>
							<td>@mdo</td>
						</tr>
						<tr>
							<td>2</td>
							<td>Jacob</td>
							<td>Thornton</td>
							<td>sd</td>
							<td>@fat</td>
						</tr>
						<tr>
							<td>3</td>
							<td>Larry the Bird</td>
							<td>@twitter</td>
							<td>@twitter</td>
							<td>sd</td>
						</tr>
					</tbody>
				</table>
			</div>
		);
  }
}

OrderTable.propTypes = {
  actions: PropTypes.object.isRequired,
  currentPair: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
	currentPair: state.currentPair
  };
}

function mapDispatchToProps(dispatch) {
  return {
	actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(OrderTable);
