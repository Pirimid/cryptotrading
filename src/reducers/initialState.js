export default {
    currentPair: {
        unit1: 'BTC',
        unit2: 'USD',
        lastTradePrice: 6568.01,
        last24HourVolume: 16681,
        last24HourPrice: -6.35
    },
    appData: {
        exchanges: [
            'GDAX',
            'Bitfinex',
            'Binance',
            'CEX.IO',
            'Bittrex',
            'Poloniex',
            'HitBTC',
            'Kraken'
        ],
        availablePairs: {
            pairCategory1: [
                {
                    pairCategory: 'Bitcoin (BTC)',
                    unit1: 'BTC',
                    unit2: 'USD',
                    symbolUnit2: '$',
                    lastTradePrice: 11123.01,
                    last24HourVolume: 16681,
                    last24HourPrice: -6.35
                },
                {
                    pairCategory: 'Bitcoin (BTC)',
                    unit1: 'BTC',
                    unit2: 'EUR',
                    symbolUnit2: '€',
                    lastTradePrice: 11123.01,
                    last24HourVolume: 1681,
                    last24HourPrice: +6.35
                },
                {
                    pairCategory: 'Bitcoin (BTC)',
                    unit1: 'BTC',
                    unit2: 'GBP',
                    symbolUnit2: '£',
                    lastTradePrice: 11123.01,
                    last24HourVolume: 16681,
                    last24HourPrice: -6.35
                }
            ]
        },
    },
    orderForm: {
        activeTab: 'market',
        isBuy: true,
        buyAmount: '',
        buySize: 0.00000000,
        limitPrice: 0,
        exchange: 'GDAX'
    },
    orderTable: [

    ],
    orderBook: {
        active: [
            { marketSize: 0.357077, price: 6574.215243, mySize: 2.90908766, exchange: 'GDAX' },
            { marketSize: 0.679435, price: 6599.694518, mySize: 2.90909087, exchange: 'Bitfinex' },
            { marketSize: 0.392460, price: 6574.215243, mySize: 2.90909000, exchange: 'GDAX' },
            { marketSize: 0.569231, price: 6568.929686, mySize: 0.90908766, exchange: 'Binance' },
            { marketSize: 0.352532, price: 6564.256353, mySize: 2.90909000, exchange: 'Bittrex' },
            { marketSize: 0.813735, price: 6598.563322, mySize: 2.90909087, exchange: 'Bitfinex' },
            { marketSize: 0.128852, price: 6576.003516, mySize: 2.90908766, exchange: 'Kraken' },
            { marketSize: 0.733126, price: 6581.451376, mySize: 2.90909000, exchange: 'CEX.IO' },
            { marketSize: 0.188341, price: 6574.533322, mySize: 2.90909087, exchange: 'GDAX' },
            { marketSize: 0.055060, price: 6599.843322, mySize: 2.90909099, exchange: 'Bitfinex' },
            { marketSize: 0.359570, price: 6576.053516, mySize: 2.90909022, exchange: 'Kraken' },
            { marketSize: 0.300259, price: 6564.346353, mySize: 2.90909099, exchange: 'Bittrex' },
            { marketSize: 0.185300, price: 6615.514027, mySize: 2.90909022, exchange: 'HitBTC' },
            { marketSize: 0.485538, price: 6574.313312, mySize: 2.90909099, exchange: 'GDAX' },
            { marketSize: 0.641470, price: 6568.952435, mySize: 2.90908766, exchange: 'Binance' },
            { marketSize: 0.344370, price: 6574.529851, mySize: 2.90909000, exchange: 'GDAX' }
        ],
        1: [
            { marketSize: 0.678425, price: 6588.484518, mySize: 2.80808087, exchange: 'Bitfinex' },
            { marketSize: 0.257077, price: 6574.515242, mySize: 2.80808766, exchange: 'GDAX' },
            { marketSize: 0.258570, price: 6576.052516, mySize: 2.80808022, exchange: 'Kraken' },
            { marketSize: 0.568221, price: 6568.228686, mySize: 0.80808766, exchange: 'Binance' },
            { marketSize: 0.252522, price: 6564.156252, mySize: 2.80808000, exchange: 'Bittrex' },
            { marketSize: 0.282460, price: 6574.215242, mySize: 2.80808000, exchange: 'GDAX' },
            { marketSize: 0.722126, price: 6581.451276, mySize: 2.80808000, exchange: 'CEX.IO' },
            { marketSize: 0.128852, price: 6576.002516, mySize: 2.80808766, exchange: 'Kraken' },
            { marketSize: 0.055060, price: 6588.842222, mySize: 2.80808088, exchange: 'Bitfinex' },
            { marketSize: 0.188241, price: 6574.522222, mySize: 2.80808087, exchange: 'GDAX' },
            { marketSize: 0.812725, price: 6588.562222, mySize: 2.80808087, exchange: 'Bitfinex' },
            { marketSize: 0.485528, price: 6574.212212, mySize: 2.80808088, exchange: 'GDAX' },
            { marketSize: 0.641470, price: 6568.852425, mySize: 2.80808766, exchange: 'Binance' },
            { marketSize: 0.185200, price: 6615.514027, mySize: 2.80808022, exchange: 'HitBTC' },
            { marketSize: 0.200258, price: 6564.246252, mySize: 2.80808088, exchange: 'Bittrex' },
            { marketSize: 0.244270, price: 6574.528851, mySize: 2.80808000, exchange: 'GDAX' }
        ],
        2: [
            { marketSize: 0.569241, price: 6569.929696, mySize: 0.90909866, exchange: 'Binance' },
            { marketSize: 0.689445, price: 6599.694519, mySize: 2.90909098, exchange: 'Bitfinex' },
            { marketSize: 0.458088, price: 6584.215244, mySize: 2.90909866, exchange: 'GDAX' },
            { marketSize: 0.844126, price: 6591.451486, mySize: 2.90909000, exchange: 'CEX.IO' },
            { marketSize: 0.492460, price: 6584.215244, mySize: 2.90909000, exchange: 'GDAX' },
            { marketSize: 0.914845, price: 6599.564422, mySize: 2.90909098, exchange: 'Bitfinex' },
            { marketSize: 0.452542, price: 6564.256454, mySize: 2.90909000, exchange: 'Bittrex' },
            { marketSize: 0.459580, price: 6586.054516, mySize: 2.90909022, exchange: 'Kraken' },
            { marketSize: 0.055060, price: 6599.944422, mySize: 2.90909099, exchange: 'Bitfinex' },
            { marketSize: 0.199441, price: 6584.544422, mySize: 2.90909098, exchange: 'GDAX' },
            { marketSize: 0.129952, price: 6586.004516, mySize: 2.90909866, exchange: 'Kraken' },
            { marketSize: 0.400259, price: 6564.446454, mySize: 2.90909099, exchange: 'Bittrex' },
            { marketSize: 0.444480, price: 6584.529951, mySize: 2.90909000, exchange: 'GDAX' },
            { marketSize: 0.641480, price: 6569.952445, mySize: 2.90909866, exchange: 'Binance' },
            { marketSize: 0.195400, price: 6615.514028, mySize: 2.90909022, exchange: 'HitBTC' },
            { marketSize: 0.495549, price: 6584.414412, mySize: 2.90909099, exchange: 'GDAX' }
        ],
        3: [
            { marketSize: 0.352532, price: 6564.256353, mySize: 2.80808000, exchange: 'Bittrex' },
            { marketSize: 0.382460, price: 6574.215243, mySize: 2.80808000, exchange: 'GDAX' },
            { marketSize: 0.678435, price: 6588.684517, mySize: 2.80808077, exchange: 'Bitfinex' },
            { marketSize: 0.357077, price: 6574.215243, mySize: 2.80807766, exchange: 'GDAX' },
            { marketSize: 0.733126, price: 6571.451376, mySize: 2.80808000, exchange: 'CEX.IO' },
            { marketSize: 0.568231, price: 6567.828676, mySize: 0.80807766, exchange: 'Binance' },
            { marketSize: 0.713735, price: 6587.563322, mySize: 2.80808077, exchange: 'Bitfinex' },
            { marketSize: 0.358570, price: 6576.053516, mySize: 2.80808022, exchange: 'Kraken' },
            { marketSize: 0.177341, price: 6574.533322, mySize: 2.80808077, exchange: 'GDAX' },
            { marketSize: 0.127752, price: 6576.003516, mySize: 2.80807766, exchange: 'Kraken' },
            { marketSize: 0.300258, price: 6564.346353, mySize: 2.80808088, exchange: 'Bittrex' },
            { marketSize: 0.344370, price: 6574.528751, mySize: 2.80808000, exchange: 'GDAX' },
            { marketSize: 0.055060, price: 6588.743322, mySize: 2.80808088, exchange: 'Bitfinex' },
            { marketSize: 0.175300, price: 6615.514027, mySize: 2.80808022, exchange: 'HitBTC' },
            { marketSize: 0.475537, price: 6574.313312, mySize: 2.80808088, exchange: 'GDAX' },
            { marketSize: 0.641470, price: 6567.852435, mySize: 2.80807766, exchange: 'Binance' }
        ],
        0: [
            { marketSize: 0.679425, price: 6599.694516, mySize: 2.90909067, exchange: 'Bitfinex' },
            { marketSize: 0.292460, price: 6574.215242, mySize: 2.90909000, exchange: 'GDAX' },
            { marketSize: 0.569221, price: 6566.929666, mySize: 0.90906766, exchange: 'Binance' },
            { marketSize: 0.612725, price: 6596.562222, mySize: 2.90909067, exchange: 'Bitfinex' },
            { marketSize: 0.166241, price: 6574.522222, mySize: 2.90909067, exchange: 'GDAX' },
            { marketSize: 0.126652, price: 6576.002516, mySize: 2.90906766, exchange: 'Kraken' },
            { marketSize: 0.257077, price: 6574.215242, mySize: 2.90906766, exchange: 'GDAX' },
            { marketSize: 0.252522, price: 6564.256252, mySize: 2.90909000, exchange: 'Bittrex' },
            { marketSize: 0.259570, price: 6576.052516, mySize: 2.90909022, exchange: 'Kraken' },
            { marketSize: 0.055060, price: 6599.642222, mySize: 2.90909099, exchange: 'Bitfinex' },
            { marketSize: 0.722126, price: 6561.451276, mySize: 2.90909000, exchange: 'CEX.IO' },
            { marketSize: 0.465526, price: 6574.212212, mySize: 2.90909099, exchange: 'GDAX' },
            { marketSize: 0.165200, price: 6615.514027, mySize: 2.90909022, exchange: 'HitBTC' },
            { marketSize: 0.244270, price: 6574.529651, mySize: 2.90909000, exchange: 'GDAX' },
            { marketSize: 0.641470, price: 6566.952425, mySize: 2.90906766, exchange: 'Binance' },
            { marketSize: 0.200259, price: 6564.246252, mySize: 2.90909099, exchange: 'Bittrex' }
        ],
    },
    tradeHistory: {
        active: [
            { tradeSize: 0.02279837, up: true, price: 6574.529651, time: '22:28:54', exchange: 'GDAX' },
            { tradeSize: 0.01879000, up: true, price: 6574.529451, time: '22:29:50', exchange: 'GDAX' },
            { tradeSize: 5.05579837, up: true, price: 6599.642222, time: '22:28:50', exchange: 'Bitfinex' },
            { tradeSize: 0.02279837, up: false, price: 6564.244552, time: '22:27:50', exchange: 'Bittrex' },
            { tradeSize: 0.05579837, up: false, price: 6599.642542, time: '22:27:49', exchange: 'Bitfinex' },
            { tradeSize: 0.02279837, up: true, price: 6561.451276, time: '22:26:40', exchange: 'CEX.IO' },
            { tradeSize: 0.01879000, up: false, price: 6566.952425, time: '22:27:56', exchange: 'Binance' },
            { tradeSize: 0.01879000, up: false, price: 6576.052516, time: '22:26:50', exchange: 'Kraken' },
            { tradeSize: 0.05579837, up: true, price: 6574.529541, time: '22:26:30', exchange: 'GDAX' },
            { tradeSize: 0.01279837, up: true, price: 6599.646542, time: '22:26:20', exchange: 'Bitfinex' },
            { tradeSize: 0.13879837, up: false, price: 6615.514237, time: '22:25:47', exchange: 'HitBTC' },
            { tradeSize: 0.01279837, up: false, price: 6564.246252, time: '22:25:48', exchange: 'Bittrex' },
            { tradeSize: 0.02279837, up: true, price: 6574.529251, time: '22:22:50', exchange: 'GDAX' },
            { tradeSize: 4.13879837, up: true, price: 6576.054716, time: '22:25:50', exchange: 'Kraken' },
            { tradeSize: 1.01279837, up: false, price: 6574.512651, time: '22:25:41', exchange: 'GDAX' },
            { tradeSize: 0.01279837, up: true, price: 6599.643642, time: '22:21:10', exchange: 'Bitfinex' },
            { tradeSize: 0.05579837, up: true, price: 6576.054721, time: '22:21:50', exchange: 'GDAX' },
            { tradeSize: 1.11879000, up: true, price: 6599.646547, time: '22:20:30', exchange: 'Bitfinex' },
            { tradeSize: 0.01879000, up: true, price: 6566.952145, time: '22:25:30', exchange: 'Binance' },
            { tradeSize: 0.01279837, up: false, price: 6564.241252, time: '22:21:01', exchange: 'Bittrex' },
            { tradeSize: 0.13879837, up: true, price: 6576.051476, time: '22:19:06', exchange: 'Kraken' },
            { tradeSize: 0.02279837, up: false, price: 6576.054121, time: '22:20:07', exchange: 'GDAX' },
            { tradeSize: 0.13879837, up: true, price: 6576.059636, time: '22:21:06', exchange: 'Kraken' },
            { tradeSize: 9.91279837, up: true, price: 6599.646123, time: '22:19:10', exchange: 'Bitfinex' },
            { tradeSize: 0.01279837, up: false, price: 6564.241245, time: '22:19:01', exchange: 'Bittrex' },
            { tradeSize: 0.15579837, up: true, price: 6614.522367, time: '22:19:50', exchange: 'HitBTC' }
        ],
        1: [
            { tradeSize: 5.05588838, up: true, price: 6588.642222, time: '22:28:50', exchange: 'Bitfinex' },
            { tradeSize: 0.02288838, up: true, price: 6584.528651, time: '22:28:54', exchange: 'GDAX' },
            { tradeSize: 0.02288838, up: false, price: 6564.244552, time: '22:28:50', exchange: 'Bittrex' },
            { tradeSize: 0.01888000, up: true, price: 6584.528451, time: '22:28:50', exchange: 'GDAX' },
            { tradeSize: 0.05588838, up: false, price: 6588.642542, time: '22:28:48', exchange: 'Bitfinex' },
            { tradeSize: 0.05588838, up: true, price: 6584.528541, time: '22:26:30', exchange: 'GDAX' },
            { tradeSize: 0.01888000, up: false, price: 6566.852425, time: '22:28:56', exchange: 'Binance' },
            { tradeSize: 0.01888000, up: false, price: 6586.052516, time: '22:26:50', exchange: 'Kraken' },
            { tradeSize: 0.01288838, up: false, price: 6564.246252, time: '22:25:48', exchange: 'Bittrex' },
            { tradeSize: 0.02288838, up: true, price: 6561.451286, time: '22:26:40', exchange: 'CEX.IO' },
            { tradeSize: 0.02288838, up: true, price: 6584.528251, time: '22:22:50', exchange: 'GDAX' },
            { tradeSize: 0.01288838, up: true, price: 6588.646542, time: '22:26:20', exchange: 'Bitfinex' },
            { tradeSize: 1.01288838, up: false, price: 6584.512651, time: '22:25:41', exchange: 'GDAX' },
            { tradeSize: 1.11888000, up: true, price: 6588.646548, time: '22:20:30', exchange: 'Bitfinex' },
            { tradeSize: 4.13888838, up: true, price: 6586.054816, time: '22:25:50', exchange: 'Kraken' },
            { tradeSize: 0.05588838, up: true, price: 6586.054821, time: '22:21:50', exchange: 'GDAX' },
            { tradeSize: 0.13888838, up: false, price: 6615.514238, time: '22:25:48', exchange: 'HitBTC' },
            { tradeSize: 0.01288838, up: true, price: 6588.643642, time: '22:21:10', exchange: 'Bitfinex' },
            { tradeSize: 0.01288838, up: false, price: 6564.241252, time: '22:21:01', exchange: 'Bittrex' },
            { tradeSize: 0.01888000, up: true, price: 6566.852145, time: '22:25:30', exchange: 'Binance' },
            { tradeSize: 8.81288838, up: true, price: 6588.646123, time: '22:18:10', exchange: 'Bitfinex' },
            { tradeSize: 0.02288838, up: false, price: 6586.054121, time: '22:20:08', exchange: 'GDAX' },
            { tradeSize: 0.13888838, up: true, price: 6586.058636, time: '22:21:06', exchange: 'Kraken' },
            { tradeSize: 0.15588838, up: true, price: 6614.522368, time: '22:18:50', exchange: 'HitBTC' },
            { tradeSize: 0.01288838, up: false, price: 6564.241245, time: '22:18:01', exchange: 'Bittrex' },
            { tradeSize: 0.13888838, up: true, price: 6586.051486, time: '22:18:06', exchange: 'Kraken' }



        ],
        2: [
            { tradeSize: 0.05577737, up: false, price: 6577.642542, time: '22:27:47', exchange: 'Bitfinex' },
            { tradeSize: 0.01777000, up: true, price: 6574.527451, time: '22:27:50', exchange: 'GDAX' },
            { tradeSize: 5.05577737, up: true, price: 6577.642222, time: '22:27:50', exchange: 'Bitfinex' },
            { tradeSize: 0.02277737, up: false, price: 6564.244552, time: '22:27:50', exchange: 'Bittrex' },
            { tradeSize: 0.02277737, up: true, price: 6561.451276, time: '22:26:40', exchange: 'CEX.IO' },
            { tradeSize: 0.01277737, up: true, price: 6577.646542, time: '22:26:20', exchange: 'Bitfinex' },
            { tradeSize: 0.02277737, up: true, price: 6574.527651, time: '22:27:54', exchange: 'GDAX' },
            { tradeSize: 0.01777000, up: false, price: 6576.052516, time: '22:26:50', exchange: 'Kraken' },
            { tradeSize: 0.05577737, up: true, price: 6574.527541, time: '22:26:30', exchange: 'GDAX' },
            { tradeSize: 0.01777000, up: true, price: 6566.752145, time: '22:25:30', exchange: 'Binance' },
            { tradeSize: 0.13777737, up: false, price: 6615.514237, time: '22:25:47', exchange: 'HitBTC' },
            { tradeSize: 0.01777000, up: false, price: 6566.752425, time: '22:27:56', exchange: 'Binance' },
            { tradeSize: 0.02277737, up: true, price: 6574.527251, time: '22:22:50', exchange: 'GDAX' },
            { tradeSize: 4.13777737, up: true, price: 6576.054716, time: '22:25:50', exchange: 'Kraken' },
            { tradeSize: 1.01277737, up: false, price: 6574.512651, time: '22:25:41', exchange: 'GDAX' },
            { tradeSize: 0.01277737, up: true, price: 6577.643642, time: '22:21:10', exchange: 'Bitfinex' },
            { tradeSize: 0.01277737, up: false, price: 6564.246252, time: '22:25:47', exchange: 'Bittrex' },
            { tradeSize: 1.11777000, up: true, price: 6577.646547, time: '22:20:30', exchange: 'Bitfinex' },
            { tradeSize: 0.01277737, up: false, price: 6564.241252, time: '22:21:01', exchange: 'Bittrex' },
            { tradeSize: 0.05577737, up: true, price: 6576.054721, time: '22:21:50', exchange: 'GDAX' },
            { tradeSize: 0.13777737, up: true, price: 6576.051476, time: '22:17:06', exchange: 'Kraken' },
            { tradeSize: 0.01277737, up: false, price: 6564.241245, time: '22:17:01', exchange: 'Bittrex' },
            { tradeSize: 0.13777737, up: true, price: 6576.057636, time: '22:21:06', exchange: 'Kraken' },
            { tradeSize: 0.15577737, up: true, price: 6614.522367, time: '22:17:50', exchange: 'HitBTC' },
            { tradeSize: 7.71277737, up: true, price: 6577.646123, time: '22:17:10', exchange: 'Bitfinex' },
            { tradeSize: 0.02277737, up: false, price: 6576.054121, time: '22:20:07', exchange: 'GDAX' }

        ],
        3: [
            { tradeSize: 0.02269936, up: false, price: 6564.244552, time: '22:26:50', exchange: 'Bittrex' },
            { tradeSize: 0.01969000, up: true, price: 6564.529451, time: '22:29:50', exchange: 'GDAX' },
            { tradeSize: 5.05569936, up: true, price: 6599.642222, time: '22:29:50', exchange: 'Bitfinex' },
            { tradeSize: 0.02269936, up: true, price: 6564.529651, time: '22:29:54', exchange: 'GDAX' },
            { tradeSize: 0.02269936, up: true, price: 6561.451266, time: '22:26:40', exchange: 'CEX.IO' },
            { tradeSize: 0.05569936, up: true, price: 6564.529541, time: '22:26:30', exchange: 'GDAX' },
            { tradeSize: 0.01969000, up: false, price: 6566.952425, time: '22:26:56', exchange: 'Binance' },
            { tradeSize: 0.05569936, up: false, price: 6599.642542, time: '22:26:49', exchange: 'Bitfinex' },
            { tradeSize: 0.01969000, up: false, price: 6566.052516, time: '22:26:50', exchange: 'Kraken' },
            { tradeSize: 0.01269936, up: true, price: 6599.646542, time: '22:26:20', exchange: 'Bitfinex' },
            { tradeSize: 0.13969936, up: false, price: 6615.514236, time: '22:25:46', exchange: 'HitBTC' },
            { tradeSize: 0.01269936, up: false, price: 6564.246252, time: '22:25:49', exchange: 'Bittrex' },
            { tradeSize: 0.02269936, up: true, price: 6564.529251, time: '22:22:50', exchange: 'GDAX' },
            { tradeSize: 4.13969936, up: true, price: 6566.054616, time: '22:25:50', exchange: 'Kraken' },
            { tradeSize: 1.01269936, up: false, price: 6564.512651, time: '22:25:41', exchange: 'GDAX' },
            { tradeSize: 0.01269936, up: true, price: 6599.643642, time: '22:21:10', exchange: 'Bitfinex' },
            { tradeSize: 0.01969000, up: true, price: 6566.952145, time: '22:25:30', exchange: 'Binance' },
            { tradeSize: 0.05569936, up: true, price: 6566.054621, time: '22:21:50', exchange: 'GDAX' },
            { tradeSize: 1.11969000, up: true, price: 6599.646546, time: '22:20:30', exchange: 'Bitfinex' },
            { tradeSize: 0.01269936, up: false, price: 6564.241252, time: '22:21:01', exchange: 'Bittrex' },
            { tradeSize: 0.13969936, up: true, price: 6566.059636, time: '22:21:06', exchange: 'Kraken' },
            { tradeSize: 0.02269936, up: false, price: 6566.054121, time: '22:20:06', exchange: 'GDAX' },
            { tradeSize: 0.01269936, up: false, price: 6564.241245, time: '22:19:01', exchange: 'Bittrex' },
            { tradeSize: 0.13969936, up: true, price: 6566.051466, time: '22:19:06', exchange: 'Kraken' },
            { tradeSize: 9.91269936, up: true, price: 6599.646123, time: '22:19:10', exchange: 'Bitfinex' },
            { tradeSize: 0.15569936, up: true, price: 6614.522366, time: '22:19:50', exchange: 'HitBTC' }
        ],
        0: [
            { tradeSize: 0.02286838, up: true, price: 6583.526651, time: '22:28:53', exchange: 'GDAX' },
            { tradeSize: 5.05586838, up: true, price: 6566.632222, time: '22:28:50', exchange: 'Bitfinex' },
            { tradeSize: 0.02286838, up: false, price: 6563.233552, time: '22:28:50', exchange: 'Bittrex' },
            { tradeSize: 0.05586838, up: false, price: 6566.632532, time: '22:28:36', exchange: 'Bitfinex' },
            { tradeSize: 0.02286838, up: true, price: 6561.351286, time: '22:26:30', exchange: 'CEX.IO' },
            { tradeSize: 0.01886000, up: true, price: 6583.526351, time: '22:26:50', exchange: 'GDAX' },
            { tradeSize: 0.01286838, up: true, price: 6566.636532, time: '22:26:20', exchange: 'Bitfinex' },
            { tradeSize: 0.01886000, up: false, price: 6586.052516, time: '22:26:50', exchange: 'Kraken' },
            { tradeSize: 0.05586838, up: true, price: 6583.526531, time: '22:26:30', exchange: 'GDAX' },
            { tradeSize: 0.13886838, up: false, price: 6615.513238, time: '22:25:38', exchange: 'HitBTC' },
            { tradeSize: 1.01286838, up: false, price: 6583.512651, time: '22:25:31', exchange: 'GDAX' },
            { tradeSize: 3.13886838, up: true, price: 6586.053816, time: '22:25:50', exchange: 'Kraken' },
            { tradeSize: 0.02286838, up: true, price: 6583.526251, time: '22:22:50', exchange: 'GDAX' },
            { tradeSize: 1.11886000, up: true, price: 6566.636538, time: '22:20:30', exchange: 'Bitfinex' },
            { tradeSize: 0.01886000, up: false, price: 6566.652325, time: '22:28:56', exchange: 'Binance' },
            { tradeSize: 0.01286838, up: true, price: 6566.633632, time: '22:21:10', exchange: 'Bitfinex' },
            { tradeSize: 0.05586838, up: true, price: 6586.053821, time: '22:21:50', exchange: 'GDAX' },
            { tradeSize: 0.01886000, up: true, price: 6566.652135, time: '22:25:30', exchange: 'Binance' },
            { tradeSize: 0.02286838, up: false, price: 6586.053121, time: '22:20:08', exchange: 'GDAX' },
            { tradeSize: 0.01286838, up: false, price: 6563.236252, time: '22:25:38', exchange: 'Bittrex' },
            { tradeSize: 0.13886838, up: true, price: 6586.051386, time: '22:16:06', exchange: 'Kraken' },
            { tradeSize: 0.01286838, up: false, price: 6563.231235, time: '22:16:01', exchange: 'Bittrex' },
            { tradeSize: 0.13886838, up: true, price: 6586.056636, time: '22:21:06', exchange: 'Kraken' },
            { tradeSize: 6.61286838, up: true, price: 6566.636123, time: '22:16:10', exchange: 'Bitfinex' },
            { tradeSize: 0.15586838, up: true, price: 6613.522368, time: '22:16:50', exchange: 'HitBTC' },
            { tradeSize: 0.01286838, up: false, price: 6563.231252, time: '22:21:01', exchange: 'Bittrex' }
        ],
    }

};
