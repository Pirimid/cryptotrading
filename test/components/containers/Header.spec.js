import React from "react";
import { mount, shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import {Header} from "../../../src/components/containers/Header";
import ConnectedHeader from "../../../src/components/containers/Header";
import initialState from "../../../src/reducers/initialState";

describe("<Header />", () => {
  const actions = {
    updateCurrencyPair: jest.fn()
  };
  const store = configureMockStore()(initialState);

  it("should match snapshot", () => {
    const component = create(
      <Provider store={store}>
        <ConnectedHeader />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should call updateCurrencyPair with selected pair", () => {
    const wrapper = shallow(
      <Header
        actions={actions}
        currentPair={initialState.currentPair}
        appData={initialState.appData}
      />
    );

    wrapper.find('li').first().simulate("click");
    expect(actions.updateCurrencyPair).toHaveBeenCalledWith(
      initialState.appData.availablePairs.pairCategory1[0]
    );
    
  });

});