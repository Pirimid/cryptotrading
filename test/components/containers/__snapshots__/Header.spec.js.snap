// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div
  className="nav"
>
  <div
    className="navItems navlogo-container"
  >
    <span
      className="navlogo"
    >
      Crypto Trade
    </span>
  </div>
  <div
    className="navItems pair-selection-container p-r has-dropdown"
  >
    <div
      className="pair-selection"
    >
      <div
        className="pair nav-title"
      >
        BTC
        /
        USD
      </div>
      <div
        className="selection nav-sm-title"
      >
        Select product
      </div>
    </div>
    <i
      aria-hidden="true"
      className="fa arrow fa-chevron-down"
    />
    <div
      className="nav-dropdown"
    >
      <h5
        className="nav-title"
      >
        BITCOIN(BTC)
      </h5>
      <ul
        className="nav-dropdown-list"
      >
        <li
          onClick={[Function]}
        >
          <a>
            <span>
              BTC
               / 
              USD
            </span>
            <span>
              $
               
              5568.01
            </span>
            <span
              className="down"
            >
              -6.35
              %
            </span>
          </a>
        </li>
        <li
          onClick={[Function]}
        >
          <a>
            <span>
              BTC
               / 
              EUR
            </span>
            <span>
              €
               
              4891.52
            </span>
            <span
              className="up"
            >
              6.51
              %
            </span>
          </a>
        </li>
        <li
          onClick={[Function]}
        >
          <a>
            <span>
              BTC
               / 
              GBP
            </span>
            <span>
              £
               
              4231.85
            </span>
            <span
              className="down"
            >
              -5.37
              %
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="navItems"
  >
    <div
      className="pair-selection"
    >
      <div
        className="pair nav-title"
      >
        5568.01
         
        BTC
      </div>
      <div
        className="selection nav-sm-title"
      >
        Last trade price
      </div>
    </div>
  </div>
  <div
    className="navItems"
  >
    <div
      className="pair-selection"
    >
      <div
        className="red pair nav-title"
      >
        -6.35
         %
      </div>
      <div
        className="selection nav-sm-title"
      >
        24 hour price
      </div>
    </div>
  </div>
  <div
    className="navItems"
  >
    <div
      className="pair-selection"
    >
      <div
        className="pair nav-title"
      >
        16681
         
        USD
      </div>
      <div
        className="selection nav-sm-title"
      >
        24 hour volume
      </div>
    </div>
  </div>
</div>
`;
